calico/felix/frules.py

```diff
@@ -409,7 +409,9 @@ def interface_to_chain_suffix(config, iface_name):
     :param iface_name: The interface name
     :returns string: the suffix (shortened if necessary)
     """
-    suffix = iface_name.replace(config.IFACE_PREFIX, "", 1)
-    # The suffix is surely not very long, but make sure.
-    suffix = futils.uniquely_shorten(suffix, 16)
-    return suffix
+    for prefix in config.IFACE_PREFIX:
+        if iface_name.startswith(prefix):
+            iface_name = iface_name[len(prefix):]
+            break
```

Comment:
I mean this:

       for i, prefix1 in enumerate(self.IFACE_PREFIX):
            for j, prefix2 in enumerate(self.IFACE_PREFIX):
                if j <= i:
                    continue
                if os.path.commonprefix([prefix1, prefix2]):
                    raise ConfigException(
                        "InterfacePrefix values %s and %s have a common prefix"
                        % (prefix1, prefix2)
                    )

The use case is that the frules.py code above will not work correctly if there are prefixes with a common prefix.  For example, if InterfacePrefix=tap,tapping and there is an interface named `tapping12345-ab`, the code above will give `ping12345-ab`, whereas it should give `12345-ab`.